### ISSUE TITLE
data-bs-dismiss="offcanvas" doesn't work with responsive offcanvas (eg. offcanvas-lg)

### ISSUE THREAD

**Issue by dankopi** [twbs/bootstrap#36962]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

Adding data-bs-dismiss="offcanvas" to links/buttons that should close the offcanvas only works if the offcanvas is not responsive (has class "offcanvas"). 

If you use responsive offcanvas (eg class "offcanvas-xxl"), the button doesn't do anything and requires an additional argument - data-bs-target="#id"

This makes it impossible to use anchor links to close offcanvas menus, 

### Reduced test cases

https://codepen.io/dankosan/pen/oNqQgvp

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome

### What version of Bootstrap are you using?

v5.2.0

**Comment by tungphan995** [comment]
Thank you for believing in me.

**Comment by GeoSot** [comment]
Quick fix: you can use `data-bs-target` to point the specific element. 

Explanation: The `dismiss` trigger searches for a target attribute or tries to find the closest `.modal`. As `.modal-xl` doesn't include `modal` class, it fails to recognize the instance

As a solution, we should update our docs to refer to this, and it would be nice to use responsive classes as modifiers in combination with original `modal` class

**/cc** @mdo we have discussed the modifier approach in the past (and probably is more suitable for v6)

**Comment by mdo** [comment]
Yeah we should mention this in our docs. This is kind of a complication from v5's implementation of offcanvas in the navbar if I recall—there wasn't an easy way to keep the base class, add the new dedicated responsive offcanvas component variations, and keep the base class alive. I'm still not 100% sure if/how we'll do it in v6.

**Comment by hbcondo** [comment]
> Quick fix: you can use `data-bs-target` to point the specific element.

@GeoSot, thanks for acknowledging this issue. The quick fix didn't work for me but I'm not sure what `modal` has to do with `offcanvas` in your explanation. Is it possible this issue can be resolved in v5.3?

**Comment by GeoSot** [comment]
Thank you for pointing out my mistake. I 've replaced the names, with the correct ones.


The behavior, is not broken. Dismiss triggers, work upon component classes and in theory we apply one main selector, and add modifier classes. In the specific case, `.offcanvas` class is missing as `.offcavas-xl` acts like an autonomous class (and not modifier), so js is not able to aim the proper element

**Comment by Brawl345** [comment]
Just tripped over this while switching from an offcanvas to a responsive one and the close button didn't work. Didn't notice the "data-bs-target" is in the docs in the example code here: https://getbootstrap.com/docs/5.3/components/offcanvas/#responsive

Maybe make it clearer that you have to use `data-bs-target="#myOffcanvas"` for responsive ones?

**Comment by SerratedSharp** [comment]
Yeh, this seems like something that needs a special note in the docs under the Responsive section.

**Comment by mdo** [comment]
Fixed in #42798, merged to `v6-dev`. The component is Drawer in v6. Dismiss triggers now resolve breakpoint-prefixed drawers such as `.lg:drawer`, so `data-bs-dismiss="drawer"` works without an explicit `data-bs-target`.
